File: Part 2 - Regression/Section 6 - Polynomial Regression/Polynomial_Regression/m.R
#Polynomial regression

#read data
dataset = read.csv('Position_Salaries.csv')
dataset=dataset[,2:3]

#split test and training set
# install.packages('caTools')
# library(caTools)
# set.seed(123)
# split=sample.split(dataset$Purchased,SplitRatio=0.8)
# training_set=subset(dataset,split==TRUE)
# test_set=subset(dataset,split==FALSE)

#feature scaling
#training_set[,2:3]=scale(training_set[,2:3])
#test_set[,2:3]=scale(test_set[,2:3])

#Fitting linear regression to dataset
lin_reg=lm(formula=Salary~.,data=dataset)

#Fitting polynomial regression to dataset
dataset$Level2=dataset$Level^2
dataset$Level3=dataset$Level^3
dataset$Level4=dataset$Level^4
poly_reg=lm(formula=Salary~.,data=dataset)

#visualize linear regression model
library(ggplot2)
ggplot()+
  geom_point(aes(x=dataset$Level,y=dataset$Salary),
             colour='red')+
  geom_line(aes(x=dataset$Level,y=predict(lin_reg,newdata=dataset)),
            colour='blue')+
  ggtitle('True vs bluff (Linear regression)')+
  xlab('Level')+
  ylab('Salary')

#visualize polynomial regression model
library(ggplot2)
ggplot()+
  geom_point(aes(x=dataset$Level,y=dataset$Salary),
             colour='red')+
  geom_line(aes(x=dataset$Level,y=predict(poly_reg,newdata=dataset)),
            colour='blue')+
  ggtitle('True vs bluff (Polynomial regression)')+
  xlab('Level')+
  ylab('Salary')

#predict data using linear
y_pred=predict(lin_reg,data.frame(Level=6.5))

#predict data using polynomial
y_pred1=predict(poly_reg,data.frame(Level=6.5,Level2=6.5^2,Level3=6.5^3,Level4=6.5^4))

File: Part 2 - Regression/Section 6 - Polynomial Regression/Polynomial_Regression/m.py
#polynomial regression

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
Y = dataset.iloc[:, 2].values

# Splitting the dataset into the Training set and Test set
"""from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 0.2, random_state = 0)"""

# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_Y = StandardScaler()
Y_train = sc_Y.fit_transform(Y_train)"""

#Fitting linear regression to dataset
from sklearn.linear_model import LinearRegression
lin_reg=LinearRegression()
lin_reg.fit(X,Y)


#Fitting polynomial regression to dataset
from sklearn.preprocessing import PolynomialFeatures
poly_reg=PolynomialFeatures(degree=4)
X_poly=poly_reg.fit_transform(X)
lin_reg2=LinearRegression()
lin_reg2.fit(X_poly,Y)

#Visualizing linear regression result
plt.scatter(X,Y,color='red')
plt.plot(X,lin_reg.predict(X),color='blue')
plt.title('Truth or bluff (linear regression)')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()

#Visualizing polynomial regression results
X_grid=np.arange(min(X),max(X),0.1)
X_grid=X_grid.reshape((len(X_grid),1))
plt.scatter(X,Y,color='red')
plt.plot(X_grid,lin_reg2.predict(poly_reg.fit_transform(X_grid)),color='blue')
plt.title('Truth or bluff (polynomial regression)')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()

#Predict with linear regression
Y_pred=lin_reg.predict(6.5)


#Predict with polynomial regression
Y_pred1=lin_reg2.predict(poly_reg.fit_transform(6.5))
